assets/src/apps/delivery/layouts/deck/DeckLayoutFooter.tsx

```diff
@@ -59,7 +59,10 @@ export const handleValueExpression = (
           const ownerActivity = currentActivityTree?.find(
             (activity) => !!activity.content.partsLayout.find((p: any) => p.id === lstVar[1]),
           );
-          value = value.replace(`${item}`, `{${ownerActivity.id}|${modifiedValue}}`);
+          //ownerActivity is undefined for app.spr.adaptivity.something i.e. Beagle app variables
+          if (ownerActivity) {
```

Comment:
this too is a prime example of why we need unit tests for this method. lets make sure we add all the different types of input and match with expected output